core/src/main/java/bio/terra/pearl/core/service/kit/KitRequestService.java

```diff
@@ -98,27 +102,49 @@ public static PepperKitAddress makePepperKitAddress(Profile profile) {
     /**
      * Fetch all kits for an enrollee.
      */
-    public Collection<KitRequest> getKitRequests(AdminUser adminUser, Enrollee enrollee) {
+    public Collection<KitRequest> getKitRequests(Enrollee enrollee) {
         return dao.findByEnrollee(enrollee.getId());
     }
 
+    /**
+     * Fetch all sample kits for a study environment.
+     */
+    public Collection<KitRequest> getSampleKitsForStudyEnvironment(StudyEnvironment studyEnvironment) {
+        var allKitTypes = kitTypeDao.findAll();
+        var kitTypeMap = allKitTypes.stream().collect(Collectors.toMap(KitType::getId, Function.identity()));
+        var kits = dao.findByStudyEnvironment(studyEnvironment.getId());
+        kits.forEach(kit -> {
+            kit.setKitType(kitTypeMap.get(kit.getKitTypeId()));
+            try {
+                kit.setPepperStatus(objectMapper.readValue(kit.getDsmStatus(), PepperKitStatus.class));
```

Comment:
Hmm... good question. This is probably the result of things changing during development that I didn't revisit during clean-up.

Thinking through changing this...

I could claim that this is protecting type safety in the UI code. Client-side, this is going to be deserialized to a [`PepperKitStatus`](https://github.com/broadinstitute/juniper/blob/1dd13e483c8ba1687655405dd3471a37e06914fe/ui-admin/src/api/api.tsx#L162). I'm not sure what would happen if the JSON didn't match that shape. Typescript won't care because we don't have the type information at runtime (which brings into question the validity of type safety here).

Alternatively, we could pass through `pepperStatusJson` and make the parsing explicit in the UI. That has the benefit of there being a natural place to handle unexpected JSON.

No matter what, I stand by the comment that we should be defensive about unexpected JSON from Pepper and make sure that Juniper degrades gracefully instead of being completely broken.

@devonbush, which approach do you prefer?

Reply:
If we expect bad json from Pepper (and we're not filtering it out on receipt) then we'll want to surface that in the UI, and you're right that that means initially treating the raw field as "unknown" in Typescript until proven otherwise so that our JS doesn't error out.  I lean towards keeping our backend APIs simple "here's what's in the database" and letting our frontend do validation/parsing so that we get maximum visibility into what's going on, and it's faster to iterate on and debug.  But either way works